src/Workspaces/Core/Portable/Storage/SQLite/v2/SQLiteConnectionPoolService.cs

```diff
@@ -0,0 +1,159 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Composition;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.ErrorReporting;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Shared.Utilities;
+using Microsoft.CodeAnalysis.SQLite.v2.Interop;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.SQLite.v2
+{
+    [Export]
+    [Shared]
+    internal sealed class SQLiteConnectionPoolService : IDisposable
```

Comment:
would prefer hiding this behind an interface (our general pattern).  thoughts?

Reply:
Interfaces come with an abstraction cost that is paid for my amortized reuse. There would not be any benefit here.

Reply:
The types are intentionally coupled. The consumable export area is by definition the exposed API of the implementation type.

Reply:
📝 I agree with interfaces when they are added for decoupling purposes, but here is unique since the service was added to the catalog with intentional coupling to other type(s) in the same component, and not for other use.

Reply:
Ok :)